### ISSUE TITLE
Ensure metaphor Apps work with Github installation

### ISSUE THREAD

**Issue by 6za** [kubefirst/kubefirst#253]
Update installer to allow metaphor to be installed on github mode.

**Comment by 6za** [comment]
Current PR: https://github.com/kubefirst/kubefirst/pull/251
Only add metaphor on `gitlab` installations. 

We need to ensure we can also re-use metaphor command for github installations.

**Comment by 6za** [comment]
https://github.com/kubefirst/gitops-template/pull/90

**Comment by 6za** [comment]
Today, the metaphor application are not enabled on github mode, the goal here is to do feature parity of [gitlab](https://github.com/kubefirst/metaphor-template/blob/main/.gitlab-ci.yml) actions are migrated to [github actions](https://docs.github.com/en/actions/migrating-to-github-actions/migrating-from-gitlab-cicd-to-github-actions)

There is deisgn difference, as the github runner is on github public, there will be extra steps need on github install to add any credential material to github side, to allow `metaphors` repo to trigger [Argo Workflow actions](https://github.com/marketplace/actions/submit-argo-workflows-from-github) from the github runner side.

**Comment by 6za** [comment]
Correcting the new design is that [A cluster hosted runner](https://github.com/actions-runner-controller/actions-runner-controller) will be pulling events in order to allow actions to happen. And it will communicate back results to github.

**Comment by 6za** [comment]
Actions needed: 
- Add [metaphor components](https://github.com/kubefirst/gitops-template/tree/main/components) for [setup of runners](https://github.com/actions-runner-controller/actions-runner-controller#runner-labels) per repository. 
> To centralize, all this metaphor apps setup. Today is spread on development/staging/production.

- Migrate the jobs to run on github action(yaml translation)
- Verify if any parameter used is not present on `github` or need to be fixed.

**Comment by 6za** [comment]
Github runner is working as expected: 
![image](https://user-images.githubusercontent.com/53096417/188004435-33e8355b-9e74-4a89-b41d-b7e655db9704.png)


Action sent to cluster to be executed by runner:

![image](https://user-images.githubusercontent.com/53096417/188004553-4a2f5889-0d2e-4657-be89-813825a2ac1f.png)

**Comment by 6za** [comment]
Result sent back: 
![image](https://user-images.githubusercontent.com/53096417/188004704-b5f3345f-f982-4cf7-9e9c-5fb01df55795.png)

**Comment by 6za** [comment]
To be investigated: 
![image](https://user-images.githubusercontent.com/53096417/188005227-7038a427-9272-4534-b77e-849e37adcd3f.png)

> runner seems to give up, after 1 job execution.

**Comment by 6za** [comment]
We need to update the token powers: 
![image](https://user-images.githubusercontent.com/53096417/188143696-ef23b6c0-9f86-4860-a034-d584ba035682.png)

**Comment by 6za** [comment]
Fixed issue with runner: https://github.com/kubefirst/gitops-template/commit/4b8c26027429361987d75283975c440a7cd0e7bd

> `Runner` type is not supported anymore.

**Comment by 6za** [comment]
Status of updates of metaphor, following issues found:
- #335 
- #334 
- #332 
- #331
- #330 

So, the main difference is how github is used, as been SSH, most of functions around git is expecting user/pwd HTTPS login. 

Also, ECR Registry is missing on github.

Several adjustments on the runner specs.

**Comment by 6za** [comment]
- #337

**Comment by 6za** [comment]
#213  is been worked under this ticket general objective. To have metaphors on github installs.

**Comment by 6za** [comment]
Waiting:
 -  #393

**Comment by 6za** [comment]
GITLAB string: 
![image](https://user-images.githubusercontent.com/53096417/190288667-e1ec0103-5158-442b-9706-82f6bc2150aa.png)

It doesn't work on as expected to skip CI on github action. It must be in lower case.

**Comment by jarededwards** [comment]
> GITLAB string: ![image](https://user-images.githubusercontent.com/53096417/190288667-e1ec0103-5158-442b-9706-82f6bc2150aa.png)
> 
> It doesn't work on as expected to skip CI on github action. It must be in lower case.

@6za gitlab doesnt honor uppercase, lets lowercase the commit `CI SKIP` --> `ci skip` and it should be honored by both git providers

https://docs.github.com/en/actions/managing-workflow-runs/skipping-workflow-runs

**Comment by 6za** [comment]
thanks!! yep, lower cases are working.

**Comment by 6za** [comment]
stress testing it: 
![image](https://user-images.githubusercontent.com/53096417/190289386-1dad1ea4-b803-45f5-880b-86af111546de.png)

**Comment by 6za** [comment]
Current job setup, migrate from `steps` to separated `jobs` and `needs` rule to ensure order. 
![image](https://user-images.githubusercontent.com/53096417/190289715-6c42ea9e-9456-4469-90fe-9e67855273a9.png)

**Comment by 6za** [comment]
In case of multiple `updates` on main.. it will honor the order of the commits, but the jobs will be queue once the previous one in the workflow is completed. Creating a weird effect that.. 

In 4 commits queue, there will be executed on something like:
- commit 1: publish
- commit 2: publish
- commit 3: publish
- commit 4: publish
- commit 1: development
- commit 2: development
- commit 3: development
- commit 4: development
- commit 1: staging
- commit 2: staging
- commit 3: staging
- commit 4: staging
- commit 1: release
- commit 2: release
- commit 3: release
- commit 4: release

We need to check later, if this may produce any unwanted effect.

**Comment by 6za** [comment]
- [X] Metaphor is working as expected from `main` metaphor repo: 
<img width="1755" alt="image" src="https://user-images.githubusercontent.com/53096417/190493031-51b8e772-e66f-476b-9c97-0120128ced84.png">
- [X] Metaphor-go is also working as expected
<img width="1782" alt="image" src="https://user-images.githubusercontent.com/53096417/190493230-74da368d-003f-4c33-a70f-f8306f544f43.png">

> It was detected that `go` sample, doesn't has a `build` step from branch checks before merge. 
- [X]  Metaphor frontend also working as expcted:
<img width="1789" alt="image" src="https://user-images.githubusercontent.com/53096417/190493435-9f12b17d-ceee-4c18-97bd-d8cb463bf70d.png">

**Comment by 6za** [comment]
I am still facing the issue, after #386. 
```bash
metaphor-publish-7fbd8ce1f9947e8cf9d01072a0f5cee5dndlk6-3753879044: Error: looks like "https://chartmuseum.6zray.kubefast.com/" is not a valid chart repository or cannot be reached: Get "https://chartmuseum.6zray.kubefast.com/index.yaml": dial tcp: lookup chartmuseum.6zray.kubefast.com on 172.20.0.10:53: no such host
metaphor-publish-7fbd8ce1f9947e8cf9d01072a0f5cee5dndlk6-3753879044: Error: couldn't load repositories file (/root/.config/helm/repositories.yaml): open /root/.config/helm/repositories.yaml: no such file or directory
metaphor-publish-7fbd8ce1f9947e8cf9d01072a0f5cee5dndlk6-3753879044: Usage:
metaphor-publish-7fbd8ce1f9947e8cf9d01072a0f5cee5dndlk6-3753879044:   helm push [flags]
metaphor-publish-7fbd8ce1f9947e8cf9d01072a0f5cee5dndlk6-3753879044: 
metaphor-publish-7fbd8ce1f9947e8cf9d01072a0f5cee5dndlk6-3753879044: Flags:
metaphor-publish-7fbd8ce1f9947e8cf9d01072a0f5cee5dndlk6-3753879044:       --access-token string             Send token in Authorization header [$HELM_REPO_ACCESS_TOKEN] 
```

**Comment by 6za** [comment]
#386 fixed, an edge case was found #424. But that is not blocker, it seems to be a argo level bug. 

I discussed details with @pagottoo, no other ideias on how to address #424 at the moment.
